Assets/Scripts/Layers/Adapters/DataTypeAdapters/ObjImportAdapter.cs

```diff
@@ -51,16 +52,44 @@ private string imgfilename
 
         private ObjImporter.ObjImporter importer;
 
-        public void ParseFiles(string value)
+        public bool Supports(LocalFile localFile)
         {
-            Debug.Log("receiveid files: " + value);
+            var hasObjExtention = localFile.SourceUrl.EndsWith(".obj");
+            if(!hasObjExtention)
+                return false;
 
-            if (value == "") //empty string received, so no selection was made
+            // Streamread and check if the obj at least has 3 newlines
+            using StreamReader reader = new StreamReader(localFile.LocalFilePath);
```

Comment:
I would probably remove the check in the supports to make that method leaner and if really needed, move that into Execute for performance reasons

Reply:
Only url extention check .obj (no need for lines)